### PR TITLE
Calib service scaffolding

### PR DESCRIPTION
**Pull request by cquiroz**
In this PR I add a new project that will run the calibration-service in a separate worker process in heroku

* A single instance can run at anytime
* It will share the same environment than odb thus able to use the same  db and have access to the service token
* At the moment it just listen for project events and  reads the observation configurations

It wont be enabled in heroku just yet but it was tested in a separate app

### CODE REVIEW

File: modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  header = MainParams.Header
+) {
+
+  override def main: Opts[IO[ExitCode]] =
+    Opts(serve)
+
+  lazy val serve: IO[ExitCode] = {
+    given Logger[IO] = Slf4jLogger.getLoggerFromName[IO]("calibrations-service")
+
+    import natchez.Trace.Implicits.noop
```
Comment (cquiroz) [inline modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala:67]: I'm not sure if we'd need tracing here.

File: modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+              calibrationsService.recalculateCalibrations(elem.programId)
+            }.compile.drain.start.void)
+    } yield ()
+
+  def services[F[_]: Concurrent: UUIDGen: Trace: Logger](
+    user: Option[User],
+    enums: Enums
+  )(pool: Session[F]): F[Services[F]] =
+    user match {
+      case Some(u) if u.role.access === Access.Service =>
```
Comment (cquiroz) [inline modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala:145]: We either have a valid user or we quit

File: modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  )(using L: Logger[F]): F[Unit]
+
+}
+
+object CalibrationsService {
+  def instantiate[F[_]: FlatMap](using Services[F]): CalibrationsService[F] =
+    new CalibrationsService[F] {
+      def recalculateCalibrations(pid: Program.Id)(using L: Logger[F]): F[Unit] =
+        for {
+          gnls   <- session.execute(Statements.SelectGmosNorthLongSlitConfigurations)(pid)
```
Comment (cquiroz) [inline modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala:35]: For now we just need the configurations, but we need to add in the future the actual work

File: modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala (hunk 1 above)
Comment (tpolecat) [reply modules/calibrations/src/main/scala/lucuma/odb/calibrations/Main.scala:67]: I think this is fine.

**Review by tpolecat — APPROVED** [review]
This looks reasonable. I will probably do something similar for approved configurations and may have more comments then. 👍🏻

**Comment by cquiroz** [comment]
Would we want to run everything as a separate worker or one worker with multiple tasks (calibrations, approved configurations, checks, etc...)

**Comment by tpolecat** [comment]
> Would we want to run everything as a separate worker or one worker with multiple tasks (calibrations, approved configurations, checks, etc...)

It makes sense to me to have multiple small workers, but I haven't thought about it that much. Seems simpler to me.

**Pull request merged**